tests/integration/golang/mlflow/experiment/search_test.go

```diff
@@ -1,3 +1,255 @@
+)
+
+type SearchExperimentsTestSuite struct {
+	suite.Suite
+	client   *helpers.HttpClient
+	fixtures *fixtures.ExperimentFixtures
+}
+
+func TestSearchExperimentsTestSuite(t *testing.T) {
+	suite.Run(t, new(SearchExperimentsTestSuite))
+}
+
+func (s *SearchExperimentsTestSuite) SetupTest() {
+	s.client = helpers.NewHttpClient(os.Getenv("SERVICE_BASE_URL"))
+	fixtures, err := fixtures.NewExperimentFixtures(os.Getenv("DATABASE_DSN"))
+	assert.Nil(s.T(), err)
+	s.fixtures = fixtures
+}
+func GetExperimentNames(experiments []*response.ExperimentPartialResponse) []string {
+	names := make([]string, len(experiments))
+	for i, exp := range experiments {
+		names[i] = exp.Name
+	}
+	return names
+}
+
+func GetSearchExperimentsResponse(s *SearchExperimentsTestSuite, request request.SearchExperimentsRequest) response.SearchExperimentsResponse {
+	query, err := urlquery.Marshal(request)
+	assert.Nil(s.T(), err)
+
+	resp := response.SearchExperimentsResponse{}
+	err = s.client.DoGetRequest(
+		fmt.Sprintf("%s%s?%s", mlflow.ExperimentsRoutePrefix, mlflow.ExperimentsSearchRoute, query),
+		&resp,
+	)
+	assert.Nil(s.T(), err)
+
+	return resp
+}
+
+func (s *SearchExperimentsTestSuite) Test_Ok() {
+	// 1. prepare database with test data.
+	experiments := []models.Experiment{
+		{
+			Name: "Test Experiment 1",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key2",
+					Value: "value2",
+				},
+			},
+
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name: "Test Experiment 2",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key1",
+					Value: "value1",
+				},
+			},
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name: "Test Experiment 3",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key3",
+					Value: "value3",
+				},
+			},
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name: "Test Experiment 4",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key4",
+					Value: "value4",
+				},
+			},
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name:           "Test Experiment 5",
+			Tags:           nil,
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name:           "Test Experiment 6",
+			Tags:           nil,
+			LifecycleStage: models.LifecycleStageDeleted,
+		},
+	}
+	for _, ex := range experiments {
+		_, err := s.fixtures.CreateTestExperiment(context.Background(), &models.Experiment{
+			Name: ex.Name,
+			Tags: ex.Tags,
+			CreationTime: sql.NullInt64{
+				Int64: time.Now().UTC().UnixMilli(),
+				Valid: true,
+			},
+			LastUpdateTime: sql.NullInt64{
+				Int64: time.Now().UTC().UnixMilli(),
+				Valid: true,
+			},
+			LifecycleStage:   ex.LifecycleStage,
+			ArtifactLocation: "/artifact/location",
+		})
+		assert.Nil(s.T(), err)
+	}
+	defer func() {
+		assert.Nil(s.T(), s.fixtures.UnloadFixtures())
+	}()
+
+	// Test Filter
+	resp := GetSearchExperimentsResponse(s, request.SearchExperimentsRequest{
```

Comment:
you can probably re organise this tests to be table test. in that case you will have own sub test for each particular filter and test will be a bit more readable.

Reply:
also you won't need in that case to have a `GetSearchExperimentsResponse` function.

Reply:
and this function `GetExperimentNames` will be unnecessary as well.